dbt_dry_run/test/node_runner/test_model_runner.py

```diff
@@ -78,6 +82,29 @@ def test_model_as_view_runs_create_view() -> None:
     assert node.compiled_sql in executed_sql
 
 
+def test_model_with_dbt_max_partition_variable_replaces_it() -> None:
+    mock_sql_runner = MagicMock()
+    mock_sql_runner.query.return_value = (DryRunStatus.SUCCESS, A_SIMPLE_TABLE, None)
+
+    node = SimpleNode(
+        unique_id="node1", depends_on=[], resource_type=ManifestScheduler.MODEL
+    ).to_node()
+    node.depends_on.deep_nodes = []
+    node.config.materialized = "incremental"
+    node.config.partition_by = PartitionBy(
```

Comment:
SimpleNode takes a parameter for the config so you don't need to mutate it after it is created:

```python
    node = SimpleNode(
        unique_id="node1",
        ....
        table_config=NodeConfig(materialized="incremental", partition_by=PartitionBy(field="loader_ingestion_time", data_type="timestamp")),
    ).to_node()
```